scipy/stats/tests/test_distributions.py

```diff
@@ -4629,6 +4632,160 @@ def test_burr_nan_mean_var_9544(self):
         assert_(np.isfinite(e4))
 
 
+class TestStudentizedRange:
+    # For alpha=.05, .01, and .001, and for each value of
+    # v=[1, 3, 10, 20, 120, inf], a Q was picked from each table for
+    # k=[2, 8, 14, 20].
+
+    vs = [1, 3, 10, 20, 120, np.inf]
+    ks = [2, 8, 14, 20]
+
+    # these arrays are written with `k` as column, and `v` as rows.
+    # Q values for alpha=.05 and .01 are taken from here:
+    # https://link.springer.com/chapter/10.1007/978-1-4613-9629-1_12
+    q05 = [17.97, 45.40, 54.33, 59.56,
+           4.501, 8.853, 10.35, 11.24,
+           3.151, 5.305, 6.028, 6.467,
+           2.950, 4.768, 5.357, 5.714,
+           2.800, 4.363, 4.842, 5.126,
+           2.772, 4.286, 4.743, 5.012]
+    q01 = [90.03, 227.2, 271.8, 298.0,
+           8.261, 15.64, 18.22, 19.77,
+           4.482, 6.875, 7.712, 8.226,
+           4.024, 5.839, 6.450, 6.823,
+           3.702, 5.118, 5.562, 5.827,
+           3.643, 4.987, 5.400, 5.645]
+    # Q values for alpha=.001 are taken from here:
+    # https://www.jstor.org/stable/2237810
+    q001 = [900.3, 2272, 2718, 2980,
+            18.28, 34.12, 39.69, 43.05,
+            6.487, 9.352, 10.39, 11.03,
+            5.444, 7.313, 7.966, 8.370,
+            4.772, 6.039, 6.448, 6.695,
+            4.654, 5.823, 6.191, 6.411]
+    qs = np.concatenate((q05, q01, q001))
+    ps = [.95, .99, .999]
```

Comment:
Can we move these up with `ks` and `vs`?

Reply:
The `ps` are somewhat closely associated with the Q values directly above since they correspond to the .05, .01, and .001 alpha levels, so it might make sense to keep it here. But theyre not dependent on anything so they could be moved.